### PR TITLE
Add Lucene byte vector documentation

### LINKED ISSUE #4344: [DOC] Support for Lucene Byte Sized Vector
**What do you want to do?**
 
- [ ] Request a change to existing documentation
- [x] Add new documentation
- [ ] Report a technical problem with the documentation
- [ ] Other 

**Tell us about your request.** Provide a summary of the request and all versions that are affected.
As of today, k-NN plugin only supports vectors of type float for each dimension which is 4 bytes. This is getting expensive in terms of storage especially for those use cases that requires ingestion on a large scale as we need to construct, load, save and search graphs which gets more and more costlier. There are few use cases where customers prefer to reduce memory footprints by trading off recall with a minimal loss.

Using the Lucene [ByteVector](https://lucene.apache.org/core/4_1_0/analyzers-common/org/apache/lucene/analysis/compound/hyphenation/ByteVector.html) feature, we can add support for Byte Sized Vector where each dimension of the vector is a byte integer with range [-128 to 127].

As of now, we are planning to support this feature from 2.9 and future versions of OpenSearch.

**What other resources are available?** Provide links to related issues, POCs, steps for testing, etc.
https://github.com/opensearch-project/k-NN/issues/812

### PR DESCRIPTION
**Pull request by kolchfa-aws**
### Description
Fixes #4344 

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _search-plugins/knn/knn-index.md [hunk 1]
```diff
@@ -48,14 +48,90 @@ However, if you intend to just use painless scripting or a k-NN score script, yo
...
 
-## Method Definitions
+ ### Lucene byte vector
+
+By default, k-NN vectors are `float` vectors, where each dimension is 4 bytes. If you want to save storage space, you can use `byte` vectors with the `lucene` engine. In a `byte` vector, each dimension is a signed 8-bit integer in the [-128, 127] range. 
+ 
+Byte vectors are supported only for the `lucene` engine. They are not supported for the `nmslib` and `faiss` engines.
+{: .note}
+ 
+ To use a `byte` vector, set the `data_type` parameter to `byte` when creating mappings for an index:
```
Comment (naveentatikonda) [inline _search-plugins/knn/knn-index.md:58]: Shall we also add a point here saying that `data_type` is an optional parameter and it's default value is `float` ?

Comment (naveentatikonda) [reply _search-plugins/knn/knn-index.md]: Looks good. Thanks for making the changes.

**Review by hdhalter — APPROVED** [review]
LGTM

File: _search-plugins/knn/knn-index.md [hunk 2]
```diff
@@ -48,14 +48,92 @@ However, if you intend to just use painless scripting or a k-NN score script, yo
...
+
+```json
+PUT test-index/_doc/2
+{
+  "my_vector1": [100, -128, 0]
+}
+```
+{% include copy-curl.html %}
+
+When querying, ensure to use a `byte` vector:
```
Comment (natebower) [inline _search-plugins/knn/knn-index.md:111]:
```suggestion
When querying, be sure to use a `byte` vector:
```

**Review by natebower — APPROVED** [review]
@kolchfa-aws Just one minor change. Otherwise, LGTM!

**Review by natebower — APPROVED** [review]
LGTM

**Pull request merged**